### ISSUE TITLE
modifier always equals zero

### ISSUE THREAD

**Issue by overflocat** [carlini/nn_robust_attacks#33]
I tried to use CarliniL2 to test my model. However, for most of the samples, the attacking results will be all zeros. I checked the code and I found that after I run 
`_, l, l2s, scores, nimg = self.sess.run([self.train, self.loss,
                                                         self.l2dist, self.output,
                                                         self.newimg])`
The result of the variable 'modifier' in `__init__` will always be zero. In other words, grads of the optimizer will always be zero, and it will not update the image. Therefore, I got results of all zeros.
I could make sure that my model does not include the softmax layer. Any hints?

**Comment by maxrumi** [comment]
I have same questions

**Comment by maxrumi** [comment]
Did you solve this question?I use both CarliniL2 and CarliniLi to test my model，but all the losses is zero, so it can't produce adverserial examples
